src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -245,6 +247,23 @@ private static boolean isSameRequester(ConsentArtefact consentDetail, String req
                                                 .thenReturn(consentApprovalResponse(consents)))));
     }
 
+    private Mono<Void> validateDate(List<GrantedConsent> grantedConsents){
+        Optional<GrantedConsent> falseValidation = grantedConsents.stream()
+                .filter(grantedConsent -> dateCheckerForNullAndFuture(grantedConsent) == false).findAny();
+        if(falseValidation.isPresent())
+            return Mono.error(ClientError.invalidDateRange());
+        else
+            return Mono.empty();
+    }
+
+    private Boolean dateCheckerForNullAndFuture(GrantedConsent grantedConsent){
+        if (grantedConsent.getPermission().getDateRange().getFromDate() != null &&
+                grantedConsent.getPermission().getDateRange().getFromDate().isBefore(LocalDateTime.now()) &&
+                grantedConsent.getPermission().getDateRange().getToDate() != null)
+                return true;
+        return false;
+    }
```

Comment:
- Boolean method names should start with `is`, `has` and `does` etc. Please rename the method name.
-

Reply:
Sure